src/grpc_requests/aio.py

```diff
@@ -146,27 +154,67 @@ def __del__(self):
                 pass
 
 
-def parse_request_data(reqeust_data, input_type):
-    _data = reqeust_data or {}
-    if isinstance(_data, dict):
-        request = ParseDict(_data, input_type())
-    else:
-        request = _data
-    return request
+class MessageParsersProtocol(Protocol):
+    def parse_request_data(self, request_data, input_type): ...
+
+    def parse_stream_requests(self, stream_requests_data: Iterable, input_type): ...
+
+    async def parse_response(self, response): ...
+
+    async def parse_stream_responses(self, responses: AsyncIterable): ...
```

Comment:
I'm presuming the intent here is to have `MessageParsersProtocol` function as an interface of sorts that `MessageParser` implementations will need to mirror.

If that's the case, could this base class be set up to raise NotImplementedErrors as opposed to just having the methods do no-ops?

Reply:
A protocol class is not a concrete class, so it can't be instantiated and used, so the methods are effectively no-ops

Reply:
Cool - for future posterity - I caught up on the topic by reading the [Protocols](https://typing.readthedocs.io/en/latest/spec/protocol.html#protocols) section of the type system spec. Thanks for helping me up my education.